### ISSUE TITLE
[NFR] Support Phalcon\Db\Dialect for database migrations

### ISSUE THREAD

**Issue by ruudboon** [phalcon/phalcon-devtools#307]
When using the db migration I'm missing some mysql specific fields like enum.
Probably caused by using describeColumns on the database adapter that has no support for database specific fields.

**Comment by sergeysviridenko** [comment]
@ruudboon when migration would using dialect, it won't be compatible with other databases. Could you provide more information about your suggestion?

**Comment by ruudboon** [comment]
Looking at this now I think it's something more fundamental. If it should be possible to use the migrate script to migrate from PostgreSQL to MySQL then we can only use SQL standards and shouldn't we support columns like enum or time.
In my case I'm using the migrate script to push changes from development to production where we will have the same database.

**Comment by sergeysviridenko** [comment]
@ruudboon yes, you're right generated migration must be general and must work on all databases without constraint.

**Comment by ruudboon** [comment]
@sergeysviridenko I do agree it should work on all database. 
We could convert for example time fields to a varchar so It's still "working" ?

**Comment by sergeysviridenko** [comment]
@ruudboon All fields convert to varchar if  it's common type.
